qemu/tests/blockdev_mirror_stress.py

```diff
@@ -1,24 +1,31 @@
-from virttest import utils_test
+import time
+import random
 
-from provider.blockdev_mirror_parallel import BlockdevMirrorParallelTest
+from virttest import utils_test
 
+from provider.storage_benchmark import generate_instance
+from provider.blockdev_mirror_wait import BlockdevMirrorWaitTest
 
-class BlockdevMirrorStressTest(BlockdevMirrorParallelTest):
-    """Do block-mirror and vm stress test in parallel"""
 
-    def stress_test(self):
-        """Run stress testing on vm"""
-        self.stress = utils_test.VMStress(self.main_vm, "stress", self.params)
-        self.stress.load_stress_tool()
+class BlockdevMirrorStressTest(BlockdevMirrorWaitTest):
+    """Do block-mirror with fio test as background test"""
 
-    def check_stress_running(self):
-        """stress should be running after block-mirror"""
-        if not self.stress.app_running():
-            self.test.fail("stress stopped unexpectedly")
+    def fio_thread(self):
+        fio_options = self.params.get("fio_options")
+        if fio_options:
+            self.test.log.info("Start to run fio")
+            self.fio = generate_instance(self.params, self.main_vm, 'fio')
+            fio_run_timeout = self.params.get_numeric("fio_timeout", 2400)
+            self.fio.run(fio_options, fio_run_timeout)
 
     def do_test(self):
+        bg_test = utils_test.BackgroundTest(self.fio_thread, "")
+        bg_test.start()
+        self.test.log.info("sleep random time before mirror during fio")
+        mint = self.params.get_numeric("sleep_min")
+        maxt = self.params.get_numeric("sleep_max")
+        time.sleep(random.randint(mint, maxt))
         self.blockdev_mirror()
-        self.check_stress_running()
         self.check_mirrored_block_nodes_attached()
         self.clone_vm_with_mirrored_images()
```

Comment:
before clone_vm_with_mirrored_images, it's more safe if we can kill fio by self.fio.clean(force=True)
try:
   self.blockdev_mirror()
finally:
  self.fio.clean(force=True)

Reply:
have updated, thanks @zhencliu.